programs/storage_api/src/storage_processor.rs

```diff
@@ -45,120 +120,198 @@ pub fn process_instruction(
+                    lockout_validations.resize(segments as usize, Vec::new());
+                } else {
                     return Err(InstructionError::InvalidArgument);
                 }
-
-                storage_contract.entry_height = entry_height;
-                storage_contract.hash = hash;
-
-                // move the proofs to previous_proofs
-                storage_contract.previous_proofs = storage_contract.proofs.clone();
-                storage_contract.proofs.clear();
-                storage_contract
-                    .proofs
-                    .resize(segments as usize, Vec::new());
-
-                // move lockout_validations to reward_validations
-                storage_contract.reward_validations = storage_contract.lockout_validations.clone();
-                storage_contract.lockout_validations.clear();
-                storage_contract
-                    .lockout_validations
-                    .resize(segments as usize, Vec::new());
             }
-            StorageInstruction::ProofValidation {
-                entry_height,
-                proof_mask,
-            } => {
-                if entry_height >= storage_contract.entry_height {
-                    return Err(InstructionError::InvalidArgument);
+            StorageInstruction::ClaimStorageReward { entry_height } => {
+                if keyed_accounts.len() != 1 {
+                    // keyed_accounts[1] should be the main storage key
+                    // to access its data
+                    Err(InstructionError::InvalidArgument)?;
                 }
 
-                let segment_index = get_segment_from_entry(entry_height);
-                if storage_contract.previous_proofs[segment_index].len() != proof_mask.len() {
+                if let StorageContract::ValidatorStorage {
+                    reward_validations, ..
+                } = &mut storage_contract
+                {
+                    let claims_index = get_segment_from_entry(entry_height);
+                    let _num_validations = count_valid_proofs(&reward_validations[claims_index]);
+                    // TODO can't just create lamports out of thin air
+                    // keyed_accounts[0].account.lamports += TOTAL_VALIDATOR_REWARDS * num_validations;
+                    reward_validations.clear();
+                } else if let StorageContract::ReplicatorStorage {
+                    reward_validations, ..
+                } = &mut storage_contract
+                {
+                    // if current tick height is a full segment away? then allow reward collection
+                    // storage needs to move to tick heights too, until then this makes little sense
+                    let current_index = get_segment_from_entry(tick_height);
+                    let claims_index = get_segment_from_entry(entry_height);
+                    if current_index <= claims_index || claims_index >= reward_validations.len() {
+                        println!(
```

Comment:
Can you change to `trace!` or `debug!` here.

Reply:
oh yeah that wasn't supposed to be there. I'll change it to debug. Thanks lol